pyvista/core/filters.py

```diff
@@ -4598,9 +4598,176 @@ def delaunay_2d(ugrid, tol=1e-05, alpha=0.0, offset=1.0, bound=False,
+
+        Examples
+        --------
+        Split a grid in half.
+
+        >>> import numpy as np
+        >>> import pyvista
+        >>> from pyvista import examples
+        >>> grid = examples.load_structured()
+        >>> voi_1 = grid.extract_subset([0, 80, 0, 40, 0, 1], boundary=True)
+        >>> voi_2 = grid.extract_subset([0, 80, 40, 80, 0, 1], boundary=True)
+
+        For fun, add the two grids back together and show they are
+        identical to the original grid.
+
+        >>> joined = voi_1.concatenate(voi_2, axis=1)
+        >>> assert np.allclose(grid.points, joined.points)
+        """
+        alg = vtk.vtkExtractGrid()
+        alg.SetVOI(voi)
+        alg.SetInputDataObject(dataset)
+        alg.SetSampleRate(rate)
+        alg.SetIncludeBoundary(boundary)
+        alg.Update()
+        return _get_output(alg)
+
+    def concatenate(dataset, other, axis, tolerance=0.0):
+        """Concatenate a structured grids to this grid.
+
+        Joins structured grids into a single structured grid.
+        Grids must be of compatible dimension, and must be coincident
+        along the seam. Grids must have the same point and cell data.
+        Field data is ignored.
+
+        Parameters
+        ----------
+        other : pyvista.StructuredGrid
+            Structured grid to concatenate.
+
+        axis : int
+            Axis along which to concatenate.
+
+        tolerance : float
+            Tolerance for point coincidence along joining seam.
+
+        Returns
+        --------
+        pyvista.StructuredGrid
+            Concatenated grid.
+
+        Examples
+        --------
+        Split a grid in half and join them.
+
+        >>> import numpy as np
+        >>> import pyvista
+        >>> from pyvista import examples
+        >>> grid = examples.load_structured()
+        >>> voi_1 = grid.extract_subset([0, 80, 0, 40, 0, 1], boundary=True)
+        >>> voi_2 = grid.extract_subset([0, 80, 40, 80, 0, 1], boundary=True)
+        >>> joined = voi_1.concatenate(voi_2, axis=1)
+        >>> print(grid.dimensions, 'same as', joined.dimensions)
+        [80, 80, 1] same as [80, 80, 1]
+        """
+        if axis > 2:
+            raise RuntimeError('Concatenation axis must be <= 2.')
+
+        # check dimensions are compatible
+        for i, (dim1, dim2) in enumerate(zip(dataset.dimensions,
+                                             other.dimensions)):
+            if i == axis:
+                continue
+            if dim1 != dim2:
+                raise RuntimeError('StructuredGrids with dimensions %s and %s '
+                                   'are not compatible.'
+                                   % (dataset.dimensions, other.dimensions))
+
+        # check point/cell variables are the same
+        assert set(dataset.point_arrays.keys()) == \
+               set(other.point_arrays.keys())
+        assert set(dataset.cell_arrays.keys()) == \
+               set(other.cell_arrays.keys())
```

Comment:
`assert` should only be used within CI.  There's a good SO answer here for why we should use `raise` instead:
https://stackoverflow.com/questions/20076182/what-are-the-implications-of-using-assert-vs-raise-exception